Editor/Modules/TextureModule.cs

```diff
@@ -0,0 +1,114 @@
+                //  new PropertyDefinition { type = PropertyType.Area, name = "Texture Recommendations", longName = "Recommendations for Optimizations for the Textures in the project." }
+                // above line makes stuff puke while the module is half-implemented
+            }
+        };
+
+        public override IEnumerable<IssueLayout> GetLayouts()
+        {
+            yield return k_IssueLayout;
+        }
+
+        public override void Audit(ProjectAuditorParams projectAuditorParams, IProgress progress = null)
+        {
+            // Implement your analysis here
+            var allTextures = AssetDatabase.FindAssets("t: Texture, a:assets");  // t: filter is for type of asset returned and a: filter is for location to search (only this location)
+            // Debug.Log("Locations Searched: " + String.Join(" & ", new List<string>(searchTheseFolders).ConvertAll(i => i.ToString())));
+            // Create an issue
+            var issues = new List<ProjectIssue>();
+
+
+            // Grab the texture's path/location and add it as an issue.
+            foreach (string aTexture in allTextures)    // ACTUALLY for each texture referenced....
+            {
+                var path = AssetDatabase.GUIDToAssetPath(aTexture);
+                Texture2D t = (Texture2D)AssetDatabase.LoadAssetAtPath(path, typeof(Texture2D));   // This uses the pre-grabbed Texture (string) name, & makes t reference the actual texture, (Texture2D) not a string.
+                double tSize = Profiler.GetRuntimeMemorySizeLong(t); //For display of the file size, later
+
+
+                var issue = ProjectIssue.Create(k_IssueLayout.category, t.name)
+                    .WithCustomProperties(new object[((int)TextureProperties.Num)]
+                    {
+                        t.name, //Name
+                        t.format, //Format
+                        t.dimension, //Shape
+                        t.isReadable, //Read-Write
+                        t.streamingMipmaps, //Streaming Mip Maps
+                        t.minimumMipmapLevel, // Minimum MipMap Level
+                        (t.width + "x" + t.height), // Resolution
+                        GetFileSize(tSize),
+                        path, //Location
+                    });
+
+                issues.Add(issue);
+                // issues.Add(ProjectIssue.Create(k_IssueLayout.category, ("Location of Texture: " + path)));
+
+                // add more issues...
+            }
+
+            if (issues.Count > 0)
+                projectAuditorParams.onIncomingIssues(issues);
+
+            // Notify that the analysis of this module is completed
+            projectAuditorParams.onModuleCompleted.Invoke();
+        }
+
+        private string GetFileSize(double byteCount) //Used to nicely display the file sizes in a readable format, reflecting KB, MB, GB properly
```

Comment:
can we reuse _Utils.Formatting.FormatSize_ instead?

Reply:
Updated and Adjusted to use Utils. :)